### DISCUSSION TITLE
Is it possible to have a Dev and Production Environment?

### DISCUSSION THREAD

**Discussion by Lucas-Gruber** [directus/directus#15956] [Archive]
Hello, I have a development and production environment for my project. But my Directus backend is for both environments. Is there a way to have a development directus where I can make changes/add content and sync this when I want to go live with this version?

**Comment by Thermal-Shock** [comment]
I have a similar situation, but I'm planning to handle this by adding custom statuses. 

For example - 
- anything in draft is not visible in any build.
- anything in review, approved, or published in picked up in the next test/qa build. 
- anything in approved or published is picked up by the next production build and anything approved is changed to published. 

It's then up to you to decide at what level in your content model you want to control publishing.

The "status" happens to be a nice convenience field built into the Directus field creation interface, but of course, you could have your own field called environment or something that you use for this purpose.

**Reply by rijkvanzanten** [reply]
This is my recommendation as well 🙂

**Reply by Lucas-Gruber** [reply]
Thank you for your answer, but my Problem is that not only the Content is changing. Also Schema changes with different Builds

**Reply by connorwinston** [reply]
You can export and apply schema changes between Directus instances using the CLI, https://docs.directus.io/reference/cli.html#migrate-schema-to-a-different-environment

**Reply by Thermal-Shock** [reply]
Can this method of export and apply be used to rename collections?

**Reply by connorwinston** [reply]
@azrikahar May be able to give you some more specifics on schema migration, better than I can.

**Reply by azrikahar** [reply]
> Can this method of export and apply be used to rename collections?

Technically it may work, but manually tweaking the schema yaml file does has an "at your own risk" element to it. Alternatively you may be able to able to create a [custom migration](https://docs.directus.io/extensions/migrations.html) to use [Knex's `renameTable`](https://knexjs.org/guide/schema-builder.html#renametable) to rename said table and make sure to update any references of it in relevant system tables as mentioned here: https://github.com/directus/directus/discussions/2711#discussioncomment-14800

That being said, if the goal of "renaming" here is more about the visual naming of said collection rather than it's actual table name in the database, you can view the collection in Data Model and add Collection Naming Translations:

https://user-images.githubusercontent.com/42867097/199964271-05f7452c-2a08-4131-9df6-7bc75293366a.mp4
